mlflow/anthropic/autolog.py

```diff
@@ -36,6 +41,25 @@ def patched_class_call(original, self, *args, **kwargs):
         ) as span:
             inputs = construct_full_inputs(original, self, *args, **kwargs)
             span.set_inputs(inputs)
-            outputs = original(self, *args, **kwargs)
-            span.set_outputs(outputs)
+
+            messages = [convert_message_to_mlflow_chat(msg) for msg in inputs.get("messages", [])]
+            try:
+                outputs = original(self, *args, **kwargs)
+                span.set_outputs(outputs)
+            finally:
+                # Set message attribute once at the end to avoid multiple JSON serialization
+                try:
+                    messages.append(convert_message_to_mlflow_chat(outputs))
+                    set_span_chat_messages(span, messages)
+                except Exception as e:
+                    _logger.debug(f"Failed to set chat messages for {span}. Error: {e}")
+                    raise
```

Comment:
q: why do we raise an exception for chat message logging while we don't raise for tool def?